drivers/serial/uart_ns16550.c

```diff
@@ -1237,7 +1240,8 @@ static const struct uart_driver_api uart_ns16550_driver_api = {
 	static void irq_config_func##n(const struct device *dev)              \
 	{                                                                     \
 		ARG_UNUSED(dev);                                              \
-		IRQ_CONNECT(DT_INST_IRQN(n), DT_INST_IRQ(n, priority),	      \
+		NS16550_IRQ_CONNECT(DT_INST_INTC_PARENT(n), DT_INST_IRQN(n),  \
```

Comment:
Shouldn't this just be converted to use `IRQ_INTC_CONNECT()` directly? It also looks like the first parameter is supposed to be an instance number, but here it's a `const struct device *`.

Reply:
In this approach, the IRQN doesn't carry information about the plic instance, so the `dev` is required